### PR TITLE
[full-ci] [tests-only] Added test for sending share invitation to user with permissions

### PR DESCRIPTION
**Pull request by grgprarup**
## Description
This PR adds a test for sending share invitation to user with different permissions.

Added test scenarios:
```feature
- Scenario Outline: send share invitation for a file to user with different permissions
- Scenario Outline: send share invitation for a folder to user with different permissions
```

## Related Issue

- Part of https://github.com/owncloud/ocis/issues/7947

## How Has This Been Tested?

- Locally
- :robot:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiSharingNg/shareInvitations.feature [hunk 1]
```diff
@@ -188,3 +188,193 @@ Feature: Send a sharing invitations
       | Co Owner    | folder        | FolderToShare  |
       | Uploader    | folder        | FolderToShare  |
       | Manager     | folder        | FolderToShare  |
+
+
+  Scenario Outline: send share invitation for file to user with different permissions
```
Comment (SagarGi) [inline tests/acceptance/features/apiSharingNg/shareInvitations.feature:193]:
```suggestion
  Scenario Outline: send share invitation for a file to user with different permissions
```

File: tests/acceptance/features/apiSharingNg/shareInvitations.feature [hunk 2]
```diff
@@ -188,3 +188,193 @@ Feature: Send a sharing invitations
...
+      | versions/update    |
+      | deleted/update     |
+      | permissions/update |
+      | standard/delete    |
+      | permissions/delete |
+      | deleted/delete     |
+      | permissions/deny   |
+
+
+  Scenario Outline: send share invitation for folder to user with different permissions
```
Comment (SagarGi) [inline tests/acceptance/features/apiSharingNg/shareInvitations.feature:288]:
```suggestion
  Scenario Outline: send share invitation for a folder to user with different permissions
```

**Review by SagarGi — COMMENTED** [review]
others LGTM :+1:

**Review by SagarGi — APPROVED** [review]

File: tests/acceptance/features/bootstrap/SharingNgContext.php [hunk 3]
```diff
@@ -104,6 +104,9 @@ public function userSendsTheFollowingShareInvitationUsingTheGraphApi(string $use
 			? $this->featureContext->getAttributeOfCreatedUser($rows['sharee'], 'id')
 			: $this->featureContext->getAttributeOfCreatedGroup($rows['sharee'], 'id');
 
+		$role = $rows['role'] ?? null;
+		$permission = $rows['permission'] ?? null;
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SharingNgContext.php:108]:
```suggestion
		$role = $rows['role'] ?: null;
		$permission = $rows['permission'] ?: null;
```

File: tests/acceptance/features/apiSharingNg/shareInvitations.feature [hunk 4]
```diff
@@ -188,3 +188,193 @@ Feature: Send a sharing invitations
...
+            }
+          }
+        }
+      }
+      """
+    Examples:
+      | permission         |
+      | permissions/read   |
+      | children/create    |
+      | upload/create      |
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/shareInvitations.feature:269]: maybe some combination as well?

File: tests/acceptance/features/bootstrap/SharingNgContext.php (hunk 3 above)
Comment (grgprarup) [reply tests/acceptance/features/bootstrap/SharingNgContext.php:108]: Using ternary op will be a problem when either role or permission is passed from the data table.

Comment (SagarGi) [reply tests/acceptance/features/bootstrap/SharingNgContext.php:108]: it did not work for me!

Comment (grgprarup) [reply tests/acceptance/features/bootstrap/SharingNgContext.php:108]: > not

which one??

Comment (grgprarup) [reply tests/acceptance/features/bootstrap/SharingNgContext.php:108]: > it did not work for me!

which one??

Comment (saw-jan) [reply tests/acceptance/features/bootstrap/SharingNgContext.php:108]: what will be the error?
It just because `??` checks null and we are assigning null

Comment (saw-jan) [reply tests/acceptance/features/bootstrap/SharingNgContext.php:108]: I think, there will be key doesn't exist error. maybe it is okey

File: tests/acceptance/features/apiSharingNg/shareInvitations.feature (hunk 4 above)
Comment (grgprarup) [reply tests/acceptance/features/apiSharingNg/shareInvitations.feature:269]: We can add more combinations in the next PR.

File: tests/acceptance/features/apiSharingNg/shareInvitations.feature [hunk 5]
```diff
@@ -188,3 +188,193 @@ Feature: Send a sharing invitations
...
+                }
+              }
+            }
+          }
+        }
+      }
+      """
+    Examples:
+      | permission         |
+      | permissions/create   |
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/shareInvitations.feature:362]:
```suggestion
      | permission         |
      | permissions/create |
```

File: tests/acceptance/features/apiSharingNg/shareInvitations.feature (hunk 4 above)
Comment (saw-jan) [reply tests/acceptance/features/apiSharingNg/shareInvitations.feature]: Please, do follow up :+1:

**Pull request merged**